### PR TITLE
LibCore: Port EventLoop to Windows

### PR DESCRIPTION
**Pull request by stasoid**

### CODE REVIEW

File: Libraries/LibCore/EventLoopImplementationWindows.cpp [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+size_t EventLoopImplementationWindows::pump(PumpMode)
+{
+    auto& thread_data = ThreadData::the();
+    auto& notifiers = thread_data.notifiers;
+    auto& timers = thread_data.timers;
+
+    size_t event_count = 1 + notifiers.size() + timers.size();
+    VERIFY(event_count <= MAXIMUM_WAIT_OBJECTS);
```
Comment (ADKaster) [inline Libraries/LibCore/EventLoopImplementationWindows.cpp:99]: This seems likely to trip on 'interesting' web pages. A fixme suggesting one of the workarounds from the docs might be appropriate.

> To wait on more than MAXIMUM_WAIT_OBJECTS handles, use one of the following methods:
>   - Create a thread to wait on MAXIMUM_WAIT_OBJECTS handles, then wait on that thread plus the other handles. Use this technique to break the handles into groups of MAXIMUM_WAIT_OBJECTS.
>  - Call [RegisterWaitForSingleObject](https://learn.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-registerwaitforsingleobject) or [SetThreadpoolWait](https://learn.microsoft.com/en-us/windows/win32/api/threadpoolapiset/nf-threadpoolapiset-setthreadpoolwait) to wait on each handle. The thread pool waits efficiently on the handles and assigns a worker thread after the object is signaled or the time-out interval expires.

**Review by ADKaster — COMMENTED** [review]
This looks reasonable. Two questions that can probably be resolved with FIXMEs

File: Libraries/LibCore/EventLoopImplementationWindows.cpp [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    HANDLE event_handles[event_count];
+    event_handles[0] = thread_data.wake_event.handle;
+
+    size_t index = 1;
+    for (auto& entry : notifiers)
+        event_handles[index++] = entry.key.handle;
+    for (auto& entry : timers)
+        event_handles[index++] = entry.key.handle;
+
+    DWORD result = WaitForMultipleObjects(event_count, event_handles, FALSE, INFINITE);
```
[the remark this answers is not in the document]
Comment (stasoid) [reply Libraries/LibCore/EventLoopImplementationWindows.cpp:110]: **WaitForMultipleObjectsEx**: We don't need this. I use overlapped I/O, I have no intention to use I/O completion routines or asynchronous procedure calls.

**MsgWaitForMultipleObjects**: EventLoopImplementationWindows is intended to be used only in helper processes. I think GUI should be handled with a separate EventLoopImplementationWindowsGUI class. Bringing Windows messaging into internal workings of the browser feels dirty. I feel that it should be dealt separately.

[the remark this answers is not in the document]
Comment (ADKaster) [reply Libraries/LibCore/EventLoopImplementationWindows.cpp:113]: Makes sense, thanks!

**Review by ADKaster — APPROVED** [review]

**Pull request merged**